lld/test/COFF/wholearchive-implib.s

```diff
@@ -3,9 +3,18 @@
 // RUN: llvm-lib -machine:amd64 -out:%t.lib -def:%t.dir/lib.def
 // RUN: llvm-mc -filetype=obj -triple=x86_64-windows %t.dir/main.s -o %t.main.obj
 
+// RUN: yaml2obj %S/Inputs/msvc-implib1.yaml -o %t.msvc-implib1.obj
```

Comment:
Btw, I wonder if we really need these extra tests (they're kinda ugly with three separate .yaml files) - I think the changes in `empty-section-decl.yaml` are enough to cover the changes in this PR, so I could drop the rest of the added tests.

Reply:
`empty-section-decl.yaml` changes seem enough to me as well.